executor/ddl_test.go

```diff
@@ -224,6 +225,33 @@ func (s *testSuite3) TestCreateView(c *C) {
 	// create view using prepare
 	tk.MustExec(`prepare stmt from "create view v10 (x) as select 1";`)
 	tk.MustExec("execute stmt")
+
+	// create view on union
+	tk.MustExec("drop table if exists t1, t2")
+	tk.MustExec("drop view if exists v")
+	_, err = tk.Exec("create view v as select * from t1 union select * from t2")
+	c.Assert(terror.ErrorEqual(err, infoschema.ErrTableNotExists), IsTrue)
+	tk.MustExec("create table t1(a int, b int)")
+	tk.MustExec("create table t2(a int, b int)")
+	tk.MustExec("insert into t1 values(1,2), (1,1), (1,2)")
+	tk.MustExec("insert into t2 values(1,1),(1,3)")
+	tk.MustExec("create definer='root'@'localhost' view v as select * from t1 union select * from t2")
+	tk.MustQuery("select * from v").Sort().Check(testkit.Rows("1 1", "1 2", "1 3"))
+	tk.MustExec("alter table t1 drop column a")
+	_, err = tk.Exec("select * from v")
+	c.Assert(terror.ErrorEqual(err, plannercore.ErrViewInvalid), IsTrue)
+	tk.MustExec("alter table t1 add column a int")
+	tk.MustQuery("select * from v").Sort().Check(testkit.Rows("1 1", "3 1", "<nil> 1", "<nil> 2"))
```

Comment:
I think the test case is wrong.  
It should be `tk.MustQuery("select * from v").Sort().Check(testkit.Rows("1 1", "1 3", "<nil> 1", "<nil> 2"))`
The result is different from MySQL.

```
mysql> select * from v;
+------+------+
| a    | b    |
+------+------+
| NULL |    2 |
| NULL |    1 |
|    1 |    1 |
|    1 |    3 |
+------+------+
4 rows in set (0.00 sec)

tidb> select * from v;
+---+---+
| a | b |
+---+---+
| NULL | 2 |
| 3 | 1 |
| NULL | 1 |
| 1 | 1 |
+---+---+
4 rows in set (0.00 sec)
```